### PR TITLE
It is needed to set used cache for Spring Data dynamically

### PR DESCRIPTION
**Pull request by schernolyas**
It is implementation of issue https://issues.apache.org/jira/browse/IGNITE-12582

### CODE REVIEW

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 1]
```diff
@@ -107,11 +118,38 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
 
         return super.getRepositoryMetadata(repoItf);
     }
+    /**
+     * The method tryes to identify SpEL extression
+     * @see <a href="https://docs.spring.io/spring/docs/5.0.16.RELEASE/spring-framework-reference/core.html#expressions">SpEL</a>
+     * @param expression
+     * @return
+     */
+
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: Remove empty line :)

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 2]
```diff
@@ -107,11 +118,38 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
+    private boolean looksLikeExpression(String expression) {
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: Maybe rename  to: isSpelExpression ?

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataCrudSelfExpressionTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+        assertEquals(person, repo.findById(id).get());
+
+        try {
+            repo.save(person);
+
+            fail("Managed to save a Person without ID");
+        }
+        catch (UnsupportedOperationException e) {
+            //excepted
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataCrudSelfExpressionTest.java]: You can use  `GridTestUtils.assertThrows` for asserts.

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java [hunk 4]
```diff
@@ -34,6 +34,13 @@
 @EnableIgniteRepositories
 public class ApplicationConfiguration {
 
+    @Bean
+    public CacheNamesBean cacheNames() {
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java]: Missing javadoc.

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/CacheNamesBean.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.springdata.misc;
+
+public class CacheNamesBean {
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/CacheNamesBean.java]: Missing javadoc. Please check that all methods and classes has javadoc :)

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java [hunk 6]
```diff
@@ -44,6 +49,8 @@
 public class IgniteRepositoryFactory extends RepositoryFactorySupport {
     /** Ignite instance */
     private Ignite ignite;
+    /** Spring application context */
```
Comment (smoldachev) [inline modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: Empty line needed

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java [hunk 7]
```diff
@@ -107,10 +117,36 @@ public IgniteRepositoryFactory(String springCfgPath) {
 
         Assert.hasText(annotation.cacheName(), "Set a name of an Apache Ignite cache using @RepositoryConfig " +
             "annotation to map this repository to the underlying cache.");
+//        https://stackoverflow.com/questions/11616316/programmatically-evaluate-a-bean-expression-with-spring-expression-language
+        String cacheName = null;
+        if (!looksLikeExpression(annotation.cacheName())) {
+            cacheName = annotation.cacheName();
```
Comment (smoldachev) [inline modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: One line if statements must be without brackets:

```java
if (!looksLikeExpression(annotation.cacheName()))
     cacheName = annotation.cacheName();
```

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 8]
```diff
@@ -107,11 +118,38 @@ public IgniteRepositoryFactory(String springCfgPath) {
 
         Assert.hasText(annotation.cacheName(), "Set a name of an Apache Ignite cache using @RepositoryConfig " +
             "annotation to map this repository to the underlying cache.");
+        String cacheName = null;
+        if (!looksLikeExpression(annotation.cacheName())) {
+            cacheName = annotation.cacheName();
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: One line if statements should be without brackets:

```java
if (!looksLikeExpression(annotation.cacheName()))
     cacheName = annotation.cacheName();
```

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 9]
```diff
@@ -107,11 +118,38 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
+        }
+        else {
+            SpelExpressionParser parser = new SpelExpressionParser();
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: Let's move all this code, into separate method.

**Comment by smoldachev** [comment]
Thank you for contribution! You are welcome :)

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 1 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 2 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: The method checks the expression for attributes of SpEL. You are right. Fixed.

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java (hunk 6 above)
Comment (schernolyas) [reply modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java (hunk 7 above)
Comment (schernolyas) [reply modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: fixed.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 8 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 9 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 2 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataCrudSelfExpressionTest.java (hunk 3 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataCrudSelfExpressionTest.java]: Tests in Spring Data don't use the class.  Are you sure that we need to change it ?

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java (hunk 4 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java]: fixed

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/CacheNamesBean.java (hunk 5 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/CacheNamesBean.java]: fixed

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataCrudSelfExpressionTest.java (hunk 3 above)
Comment (smoldachev) [reply modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataCrudSelfExpressionTest.java]: I think we should use it :) With `GridTestUtils.assertThrows` we can assert error message:

```java
        GridTestUtils .assertThrows(
            null,
            () -> repo.save(person),
            UnsupportedOperationException.class,
            "Use IgniteRepository.save(key,value) method instead."
        );
```

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 10]
```diff
@@ -107,12 +126,25 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
-
-        repoToCache.put(repoItf, annotation.cacheName());
+        String cacheName = executeExpression(annotation.cacheName());
+        repoToCache.put(repoItf, cacheName);
 
         return super.getRepositoryMetadata(repoItf);
     }
 
+    /**
+     *  execute a SpEL extression
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]:
```suggestion
     *  Evaluate a SpEL expression.
```

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 11]
```diff
@@ -107,12 +126,25 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
+     *
+     * @param spelExpression SpEL expression
+     * @return the result of execution of the SpEL expression
+     */
+    @NotNull private String executeExpression(String  spelExpression) {
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]:
```suggestion
    @NotNull private String evaluateExpression(String  spelExpression) {
```

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 12]
```diff
@@ -107,12 +126,25 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
+        StandardBeanExpressionResolver resolver = new StandardBeanExpressionResolver();
+        StandardEvaluationContext ec = new StandardEvaluationContext();
+        ec.setBeanResolver(new BeanFactoryResolver(ctx.getAutowireCapableBeanFactory()));
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: Looks like we don't use `ec` variable

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java [hunk 13]
```diff
@@ -34,6 +34,16 @@
 @EnableIgniteRepositories
 public class ApplicationConfiguration {
 
+    /**
+     * The bean with cache names
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java]: Need dots in the end :)

```suggestion
     * The bean with cache names.
```

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java [hunk 14]
```diff
@@ -34,6 +34,16 @@
...
+     */
+    @Bean
+    public CacheNamesBean cacheNames() {
+        CacheNamesBean bean = new CacheNamesBean();
+        bean.setPersonCacheName("PersonCache");
+        return bean;
```
Comment (smoldachev) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java]: Need empty line before.

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java [hunk 15]
```diff
@@ -107,12 +126,41 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
         return super.getRepositoryMetadata(repoItf);
     }
 
+    /**
+     *  execute a SpEL extression
+     *
+     * @param spelExpression SpEL expression
+     * @return the result of execution of the SpEL expression
+     */
+    @NotNull private String executeExpression(String  spelExpression) {
```
Comment (smoldachev) [inline modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]:
```suggestion
    @NotNull private String evaluateExpression(String  spelExpression) {
```

File: modules/spring-data-2.2/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java [hunk 16]
```diff
@@ -34,6 +34,16 @@
...
 public class ApplicationConfiguration {
 
+    /**
+     * The bean with cache names
+     */
+    @Bean
+    public CacheNamesBean cacheNames() {
+        CacheNamesBean bean = new CacheNamesBean();
+        bean.setPersonCacheName("PersonCache");
+        return bean;
```
Comment (smoldachev) [inline modules/spring-data-2.2/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java]: Empty line before :)

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 12 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: yes :-O .  Will be refactored

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java (hunk 15 above)
Comment (schernolyas) [reply modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.2/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java (hunk 16 above)
Comment (schernolyas) [reply modules/spring-data-2.2/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java]: fixed

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java (hunk 14 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/misc/ApplicationConfiguration.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 12 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 11 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 10 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 17]
```diff
@@ -53,8 +70,11 @@
      *
      * @param ignite
      */
-    public IgniteRepositoryFactory(Ignite ignite) {
+    public IgniteRepositoryFactory(Ignite ignite, ApplicationContext ctx) {
         this.ignite = ignite;
+        this.ctx = ctx;
+        this.beanFactory = new DefaultListableBeanFactory(ctx.getAutowireCapableBeanFactory());
+        this.beanExpressionContext = new BeanExpressionContext(beanFactory,null);
```
Comment (ingvard) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: code style, please add space.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 18]
```diff
@@ -63,8 +83,11 @@ public IgniteRepositoryFactory(Ignite ignite) {
      *
      * @param cfg Ignite configuration.
      */
-    public IgniteRepositoryFactory(IgniteConfiguration cfg) {
+    public IgniteRepositoryFactory(IgniteConfiguration cfg, ApplicationContext ctx) {
         this.ignite = Ignition.start(cfg);
+        this.ctx = ctx;
+        this.beanFactory = new DefaultListableBeanFactory(ctx.getAutowireCapableBeanFactory());
+        this.beanExpressionContext = new BeanExpressionContext(beanFactory,null);
```
Comment (ingvard) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: code style, please add space.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 19]
```diff
@@ -73,8 +96,11 @@ public IgniteRepositoryFactory(IgniteConfiguration cfg) {
      *
      * @param springCfgPath A path to Ignite configuration.
      */
-    public IgniteRepositoryFactory(String springCfgPath) {
+    public IgniteRepositoryFactory(String springCfgPath, ApplicationContext ctx) {
         this.ignite = Ignition.start(springCfgPath);
+        this.ctx = ctx;
+        this.beanFactory = new DefaultListableBeanFactory(ctx.getAutowireCapableBeanFactory());
+        this.beanExpressionContext = new BeanExpressionContext(beanFactory,null);
```
Comment (ingvard) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: code style, please add space.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java [hunk 20]
```diff
@@ -108,11 +134,23 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
     }
 
+    /**
+     *  evaluate the SpEL expression
+     *
+     * @param spelExpression SpEL expression
+     * @return the result of execution of the SpEL expression
+     */
+    @NotNull private String evaluateExpression(String spelExpression) {
+        return (String)resolver.evaluate(spelExpression,beanExpressionContext);
```
Comment (ingvard) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: code style, please add space.

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java [hunk 21]
```diff
@@ -108,11 +134,38 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
+    }
+
+    /**
+     * The method tryes to identify that the expression looks like SpEL extression
+     *
+     * @param expression string with a expression
+     * @return true if the string contains attributes of SpEL expression
+     * @see <a href="https://docs.spring.io/spring/docs/5.0.16.RELEASE/spring-framework-reference/core.html#expressions">SpEL</a>
+     */
+    private boolean isSpelExpression(String expression) {
```
Comment (ingvard) [inline modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: Please remove the dead code.

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java [hunk 22]
```diff
@@ -108,11 +134,38 @@ public IgniteRepositoryFactory(String springCfgPath) {
...
     }
 
+    /**
+     *  evaluate the SpEL expression
+     *
+     * @param spelExpression SpEL expression
+     * @return the result of execution of the SpEL expression
+     */
+    @NotNull private String evaluateExpression(String spelExpression) {
+        return (String)resolver.evaluate(spelExpression,beanExpressionContext);
```
Comment (ingvard) [inline modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: Code style, space.

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactoryBean.java [hunk 23]
```diff
@@ -66,19 +66,19 @@ protected IgniteRepositoryFactoryBean(Class<? extends T> repositoryInterface) {
         try {
             Ignite ignite = (Ignite)ctx.getBean("igniteInstance");
 
-            return new IgniteRepositoryFactory(ignite);
+            return new IgniteRepositoryFactory(ignite,ctx);
```
Comment (ingvard) [inline modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactoryBean.java]: space here and below.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 17 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 18 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 19 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java (hunk 20 above)
Comment (schernolyas) [reply modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java (hunk 21 above)
Comment (schernolyas) [reply modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: yes. thank you! fixed

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java (hunk 22 above)
Comment (schernolyas) [reply modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java]: fixed

File: modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactoryBean.java (hunk 23 above)
Comment (schernolyas) [reply modules/spring-data-2.2/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactoryBean.java]: fixed

**Comment by schernolyas** [comment]
New version has been produced

**Pull request closed without merging**